lib/spawn.ts

```diff
@@ -1,8 +1,57 @@
 import { Ok } from "./result.ts";
-import type { ScopeInternal } from "./scope.ts";
+import type { ScopeInternal } from "./scope-internal.ts";
 import { createTask, type NewTask } from "./task.ts";
 import type { Effect, Operation, Task } from "./types.ts";
 
+/**
+ * Run another operation concurrently as a child of the current one.
+ *
+ * The spawned operation will begin executing immediately and control will
+ * return to the caller when it reaches its first suspend
+ *
+ * @example
+ * ```js
+ * import { main, sleep, suspend, spawn } from 'effection';
+ *
+ * await main(function*() {
+ *   yield* spawn(function*() {
+ *     yield* sleep(1000);
+ *     console.log("hello");
+ *   });
+ *   yield* spawn(function*() {
+ *     yield* sleep(2000);
+ *     console.log("world");
+ *   });
+ *   yield* suspend();
+ * });
+ * ```
+ *
+ * You should prefer using the spawn operation over calling
+ * {@link Scope.run} from within Effection code. The reason being that a
+ * synchronous failure in the spawned operation will not be caught
+ * until the next yield point when using `run`, which results in lines
+ * being executed that should not.
+ *
+ * @example
+ * ```js
+ * import { main, suspend, spawn, useScope } from 'effection';
+ *
+ * await main(function*() {
+ *   yield* useScope();
+ *
+ *   scope.run(function*() {
```

Comment:
@cowboyd this looks like a mistake. Should this be yielding?

Reply:
Yes, this is outdated actually, so I was able to just remove it.